models/_generate_layers.py

```diff
@@ -496,6 +497,50 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer: LayerBase) -> str:
+  """
+  Inspects the given layer in RETURNN, extracts the super call from there, removes unwanted parameters
+  and then returns a string that is ready to be used in the super call for the _generated_layers file.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  # get the first line which contains super to get the super call
+  super_call = lines[0]  # super_call="super(ChoiceLayer, self).__init__(beam_size=beam_size, search=search, **kwargs)"
+  # only keep the part after init of super_call
+  call = super_call.split(".")[1]  # call = "__init__(beam_size=beam_size, search=search, **kwargs)"
+  assert call.startswith("__init__")
+  assert call.endswith(")")
+  # remove "__init__(" and ") from the call string
+  call_pruned = call[len("__init__("):-len(")")]  # call pruned = "beam_size=beam_size, search=search, **kwargs"
+
+  # get list of tuples for parameter with (param_name, value)
+  tup_ls = [x.split("=") for x in call_pruned.split(",")]
+
+  # ignore **kwargs when excluding
+  kwa = tup_ls[-1][0]
+  tup_ls = tup_ls[:-1]
+
+  # additionally excluded parameters
+  excl_params = ["batch_norm"]
+  # remove excluded params
+  tup_ls = ["=".join((key, value)).strip() for (key, value) in tup_ls
+            if (key.strip() not in LayerSignature.IgnoreParamNames and key not in excl_params)]
+  # reattach kwa
+  tup_ls.append(kwa.strip())
```

Comment:
Do not make it so unclear. You expect a certain `kwa`. So do that directly.
```suggestion
  tup_ls += ["**kwargs"]
```